WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/post/PostCardBuilder.kt

```diff
@@ -2,55 +2,95 @@ package org.wordpress.android.ui.mysite.cards.post
 
 import org.wordpress.android.R
 import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard
-import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.PostCardCreateFirst
-import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.PostCardDraftOrScheduled
-import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.PostCardDraftOrScheduled.PostItem
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.FooterLink
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.PostCardWithPostItems
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.PostCardWithPostItems.PostItem
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.PostCardWithoutPostItems
 import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.PostCardBuilderParams
 import org.wordpress.android.ui.mysite.cards.post.mockdata.MockedPostsData.Post
+import org.wordpress.android.ui.mysite.cards.post.mockdata.MockedPostsData.Posts
 import org.wordpress.android.ui.utils.UiString.UiStringRes
 import org.wordpress.android.ui.utils.UiString.UiStringText
 import javax.inject.Inject
 
 class PostCardBuilder @Inject constructor() {
     fun build(params: PostCardBuilderParams): List<PostCard> = mutableListOf<PostCard>().apply {
         val posts = params.mockedPostsData?.posts
-        posts?.hasPublishedPosts?.takeIf { it }?.let { add(createFirstPostCard()) }
-        posts?.draft?.takeIf { it.isNotEmpty() }?.let { add(it.createDraftPostsCard()) }
-        posts?.scheduled?.takeIf { it.isNotEmpty() }?.let { add(it.createScheduledPostsCard()) }
+        posts?.hasPublishedPosts?.takeIf { !posts.hasDraftsOrScheduledPosts() }
+                ?.let { hasPublishedPosts ->
+                    if (!hasPublishedPosts) {
```

Comment:
Suggestion (💡): Consider inverting the if condition. This way you will be getting rid of the extra `!` and having to read the condition will become simpler. I understand why you wanted to use the `!` in the first place, just so that the logic will read in the correct order, first the `createFirstPostCard(...)` and then the `createNextPostCard(...)`, I usually try to do the same. But, when the code complicates, even a bit, I prefer to sacrifice the order in favor of less complex code. In this case, it is only the `!`, which will only give you the extra overhead to read the `hasPublishedPosts` as `hasNoPublishedPosts`, but even that I would optimize... 😅 

PS: This is just merely me making conversation, please don't feel you need to change anything, the code is perfectly fine the way it is now, it is just a matter of preference and I just tried to explain why I usually prefer this other way.

Reply:
>  I understand why you wanted to use the ! in the first place, just so that the logic will read in the correct order, first the createFirstPostCard(...) and then the createNextPostCard(...)

You correctly mapped my mental model here 🙂. I did that to always first add logic for the "Create First" card and then for "Create Next" card  to be consistent everywhere. It's just a matter of preference, since I don't have a strong preference here I applied your suggestion in 5064abe.

Reply:
👋 @ashiagr !

So many thanks for applying my suggest, but please don't feel like you need to. As you said, it is a matter of preference.

The only thing I am doing here is opening a discussion for us, hoping that it might end to something golden and a way for us to not having to thing about this little details in the future, but rather mechanically (muscle memory) apply one way of doing things vs. the other. I am all about productivity, and thus it works for me to have this rules hardcoded/imprinted in my head so I don't have to put any effort next time, thus relaxing my mental overhead.

PS: I do the same with everything, for example, I struggled a lot with emojis, then I say to myself:
- DO: This is a great idea! 👍 
- DON'T: This is a great idea 👍 !
- EXPLANATION: I always put emojis after the sentence is done, with any period (`.`), or explanation marks (`!`), or question marks (`?`), or anything. This way, I no longer think about which way I prefer doing a sentence, when that involves an emoji. 😅 Too much I know, I was just trying to let you know how and why I think this way! 🤣 🤣 🤣